plugins/nanospell/plugin.js

```diff
@@ -18,769 +18,774 @@
-                htmlFilter = dataProcessor && dataProcessor.htmlFilter,
-                pathFilters = editor._.elementsPath && editor._.elementsPath.filters,
-                dataFilter = dataProcessor && dataProcessor.dataFilter,
-                removeFormatFilter = editor.addRemoveFormatFilter,
-                pathFilter = function (element) {
-                    if (element.hasClass('nanospell-typo')) {
-                        return false;
-                    }
-                },
-                removeFormatFilterTemplate = function (element) {
-                    var result = true;
-
-                    if (element.hasClass('nanospell-typo')) {
-                        result = false;
-                    }
-
-                    return result;
-                };
-
-            if (pathFilters) {
-                pathFilters.push(pathFilter);
-            }
-
-            if (dataFilter) {
-                var dataFilterRules = {
-                    elements: {
-                        span: function (element) {
-
-                            var scaytState = element.hasClass('nanospell-typo');
-
-                            if (scaytState) {
-                                delete element.name;
-                            }
-
-                            return element;
-                        }
-                    }
-                };
-
-                dataFilter.addRules(dataFilterRules);
-            }
-
-            if (htmlFilter) {
-                var htmlFilterRules = {
-                    elements: {
-                        span: function (element) {
-
-                            var scaytState = element.hasClass('nanospell-typo');
-
-                            if (scaytState) {
-                                delete element.name;
-                            }
-
-                            return element;
-                        }
-                    }
-                };
-
-                htmlFilter.addRules(htmlFilterRules);
-            }
-
-            if (removeFormatFilter) {
-                removeFormatFilter.call(editor, removeFormatFilterTemplate);
-            }
-        }
-
-    });
+	'use strict';
+
+	var maxRequest = 200;
+	var editorHasFocus = false;
+	var spell_delay = 250;
+	var spell_fast_after_spacebar = true;
+	var settings_path;
+	var state = false;
+	var lang = "en";
+	var locale = {
+		ignore: "Ignore",
+		learn: "Add To Personal Dictionary",
+		nosuggestions: "( No Spelling Suggestions )"
+	};
+	var spellcache = [];
+	var suggestionscache = [];
+	var ignorecache = [];
+
+	function cleanQuotes(word) {
+		return word.replace(/[\u2018\u2019]/g, "'");
+	};
+
+	// wordwalker definition
+
+	function WordWalker(range) {
+		var isNotBookmark = CKEDITOR.dom.walker.bookmark(false, true);
+
+		var startNode = range.startContainer;
+		var endNode = range.endContainer;
+
+		console.assert(startNode.equals(endNode));
+
+		function evaluator(node) {
```

Comment:
Can we give this a name that describes what it does (even though it's being assigned to the evaluator)? Maybe `isRootBlockText`?